modules/internal/ChapelArray.chpl

```diff
@@ -1613,12 +1613,14 @@ module ChapelArray {
       return this((...r));
     }
 
-    pragma "no doc"
+    /* Return an array slice halting if the range contains elements
+       that are not local. */
     proc localSlice(r: range(?)... rank) {
       return _value.dsiLocalSlice(chpl__anyStridable(r), r);
     }
 
-    pragma "no doc"
+    /* Return an array slice halting if the domain contains elements
+       that are not local. */
```

Comment:
In the current wording, I'd put a comma after "slice".

The more important aspect of this routine isn't that it halts if the slice is out-of-bounds -- it's that it returns a local view of the sub-array which is cheaper to index in subsequent operations because the indices are known to be local (so there are no checks to see which locale owns the indices, whether communication is required, etc.)

In this routine and the previous, I worry that the term "slice" may not be sufficiently well-known that these descriptions will be clear to someone who doesn't already know what the routines do.